cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+package images
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	urlP "net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v3/cmd/podman/common"
+	"github.com/containers/podman/v3/cmd/podman/parse"
+	"github.com/containers/podman/v3/cmd/podman/registry"
+	"github.com/containers/podman/v3/cmd/podman/system/connection"
+	"github.com/containers/podman/v3/libpod/define"
+	"github.com/containers/podman/v3/pkg/domain/entities"
+	"github.com/docker/distribution/reference"
+	scpD "github.com/dtylman/scp"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	saveScpDescription = `Securely copy an image from one host to another.`
+	imageScpCommand    = &cobra.Command{
+		Use:               "scp [options] IMAGE [IMAGE...]",
+		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Long:              saveScpDescription,
+		Short:             "securely copy images",
+		RunE:              scp,
+		Args:              cobra.MinimumNArgs(1),
```

Comment:
From https://github.com/containers/podman/pull/10828#discussion_r672671719 :

Non-blocking: Shouldn’t this prevent `len(args) > 2`?

(If `RangeArgs` throws “strange errors”, that should be investigated, or implemented another way, instead of giving up on the goal.)

`parseArgs` does fail on `len(args) > 2`, so this ends up working fine; it’s just not obvious here, and I would expect `RangeArgs` to work if `MinimumNArgs` works.

Reply:
I talked to @ashley-cui about this, I think `RangeArgs` might be broken in the system tests, which we can investigate. if you use range args, the image system tests pass 4 arguments no matter what and cause an error. Separate issue I believe.

Reply:
@cdoern Are you talking about the completion system test? The reason it is failing is because the use line does not match the cobra `Args` function.  If you use `RangeArgs(1,2)` the use line should be `scp [options] IMAGE IMAGE`.

Reply:
@Luap99 AH!!!! thank you, that makes SO much sense

Reply:
for my case since we only take one name, would it be `scp [options] IMAGE` @Luap99 ?

Reply:
It depends on what this commands does. Right now it accepts two arguments so your use line should show two. If the second argument is optional it should be `scp [options] IMAGE [IMAGE]`. Or even better something like this `scp [options] IMAGE [HOST::IMAGE]` I do not know what the valid syntax is, I have not followed the discussion.

Reply:
ok cool thanks, the second one is optional but is never another image so I will do `[HOST::]`

Reply:
Remember to change it in the man page as well.